drivers/disk/ramdisk.c

```diff
@@ -142,6 +142,23 @@ static const struct disk_operations ram_disk_ops = {
 		.buf = UINT_TO_POINTER(DT_REG_ADDR(DT_INST_PHANDLE(n, ram_region))),	\
 	}
 
+#define RAMDISK_DEVICE_CONFIG_DEFINE_DISKIMG(n)                                         \
+__asm__(                                                                                \
```

Comment:
IMHO, this should be done with modification to linker scripts, and linker should take care of this.

Reply:
Linker scripts are, in general, even less portable and more cumbersome from usability perspective, compared to inline assembly, unless combined with sufficient external tooling. Zephyr does have a rich framework available for leveraging linker scripts so it can indeed be leveraged to some extent. I was aware of it but hadn't explored it yet so given general concerns with linker scripts, I had gone with an entirely inline assembler-based approach. Your comment got me thinking about it so I have reworked the chaneset to take advantage of Zephyr linker scripts framework.

However, .incbin is still being used minimally. Technically, we can get rid of it too but the alternative on linker scripts side - TARGET/INPUT directives of GNU ld - would be more GNU-specific than .incbin, and other GNU linker syntax that is already being used across Zephyr source tree. It will also require adding some additional extensions to existing Zephyr linker scripts machinery, primarily because TARGET/INPUT directives must precede the site of embedding and must also be placed outside of SECTIONS block. AFAIK, existing linker scripts machinery does not have any such provision. It can of course be added, but given this rarely used feature of GNU ld, I am hesitant to rely on it. Let me know how you see it.

Reply:
OK, so I was not clear enough: what I meant is that we have macros like ITERABLE_SECTION_RAM that allows you to put objects into sections defined with help of linker.
What I tried to say is maybe we need  macro like, don't stick to naming it is example,  `BINARY_BLOB_SECTION(section, name, path`) that can be used to just inject bin into section instead of directly doing the asm here.

Reply:
yeah this code needs to go out of here, it has no place in the .c file and completely ignores that we linker generator **AND** cmake generator for these, so this would completely break usage for anyone using the cmake option

Reply:
@nordicjm This implementation has already been updated. But I am not sure about what you mean by:
> ... so this would completely break usage for anyone using the cmake option

I guess you are referring to building using CMake directly, instead of building via west? i.e. something like:
```
cmake -Bbuild -GNinja -DBOARD=reel_board samples/hello_world
ninja -Cbuild
```
instead of:
```
west build -b reel_board samples/hello_world
```
Or are you referring to something else?
If it is just building via cmake instead of west, then I don't see any problem, at least in my testing - it builds and runs fine.

Reply:
Unrelated to that, you can use LD templates (the default) or cmake generators, build with `-DCONFIG_CMAKE_LINKER_GENERATOR=y` which uses cmake files, e.g. https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/linker_script/common/common-rom.cmake

Reply:
@nordicjm Thanks for the pointer - I hadn't come across CMake linker generators yet in my exploration of Zephyr internals. I have tested with this now but again the approach in this changeset did work in both cases - i.e. with LD templates or with CMake linker generator. But a closer look indicates that I see no problem because I am still using GCC toolchains/GNU ld in both cases.

One problem I have noted is the direct use of .section/,global/.type directives. Zephyr's toolchain/linker abstraction would break here if I'd use some other toolchain that does not support this syntax and/or which (by default) does not recognize the section name used.

I have now updated the changeset to use the wrappers Zephyr toolchain abstraction provides for this purpose. To do so, I had to do away with use of inline assembly, however, because the relevant wrapper macros provided by Zephyr toolchain abstraction are not usable in inline assembler blocks. So now, equivalent assembly code is generated in a separate complementary assembly file that would be automatically included in the build, as needed (when GNU extensions are available).

Please have another look and share your thoughts.